### PR TITLE
Add exists and notExists methods SolrQueryBuilder

### PR DESCRIPTION
**Pull request by ke4**
This change adds 2 methods to the `SolrQueryBuilder` class.
- exists(schemaField) : add a query clause as a filter to the query builder and the result contains only Solr documents where the given field exists.
- notExists(schemaField) : add a query clause as a filter to the query builder and the result contains only Solr documents where the given field does not exist.

Manual checks

- [ ] This branch is passing the CI tests as part of a pull request in the atlas-web-bulk or atlas-web-scxa repos and I have indicated that PR here.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/atlas/solr/cloud/search/SolrQueryBuilder.java [hunk 1]
```diff
@@ -122,6 +124,16 @@ public <U extends SchemaField<T>> SolrQueryBuilder<T> addNegativeFilterFieldByTe
         return this;
     }
 
+    public <U extends SchemaField<T>> SolrQueryBuilder<T> exists(U fieldNameToExist) {
```
Comment (lingyun1010) [inline src/main/java/uk/ac/ebi/atlas/solr/cloud/search/SolrQueryBuilder.java]: I think it is a bit English grammer issue, `fieldNameToBeContained` would make more sense to me and also `filedNameToBeNotContained`.

Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/solr/cloud/search/SolrQueryBuilder.java]: Also for the function name, would be better if say `contain` or `notContain`.

Comment (ke4) [reply src/main/java/uk/ac/ebi/atlas/solr/cloud/search/SolrQueryBuilder.java]: @lingyun1010 I used `exists` to make it similar to Solr's same function. Please have a look at the documentation:
https://solr.apache.org/guide/solr/latest/query-guide/function-queries.html#exists-function
Thanks

Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/solr/cloud/search/SolrQueryBuilder.java]: > I think it is a bit English grammer issue, `fieldNameToBeContained` would make more sense to me and also `filedNameToBeNotContained`.

I see your points to use `exists` and `notExists` as the function name, but the varialbe names still not in a right English grammer I believe, you may ask for some other native speakers to double check.

Comment (ke4) [reply src/main/java/uk/ac/ebi/atlas/solr/cloud/search/SolrQueryBuilder.java]: Ok, I reviewed it and the name of these field are not correct, I agree. I changed them to just `filed` as they are not a `String` type, but `SchemaField` and they don't need anything to add to it as the name of the method is clear enough.
Thanks for your feedback to make this cleaner. 👍

**Review by lingyun1010 — APPROVED** [review]
LGTM!

**Pull request merged**